src/sample-account-setup.test.ts

```diff
@@ -1,27 +1,546 @@
-import { Patient } from '@medplum/fhirtypes';
-import { MockClient } from '@medplum/mock';
-import { expect, test } from 'vitest';
-import { handler } from './sample-account-setup';
+import { BotEvent, MedplumClient, createReference, getDisplayString } from '@medplum/core';
+import {
+  CarePlan,
+  DiagnosticReport,
+  Immunization,
+  Communication,
+  MedicationRequest,
+  Observation,
+  Patient,
+  Practitioner,
+  Reference,
+  Schedule,
+  Slot
+} from '@medplum/fhirtypes';
 
-const contentType = 'application/fhir+json';
-// To run this test only: npm t -- src/sample-account-setup.test.ts
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<any> {
+  const patient = event.input as Patient;
 
-test('Create data for patient', async () => {
-  const medplum = new MockClient();
+  const date = new Date();
 
-  const input = await medplum.createResource<Patient>({
-    resourceType: 'Patient',
-    name: [{ given: ['Michael'], family: 'Scott' }],
-    telecom: [
+  const getPreviousDay = (date = new Date()): Date => {
+    const previous = new Date(date.getTime());
+    previous.setDate(date.getDate() - 1);
+
+    return previous;
+  };
+
+  const subject: Reference<Patient> = {
+    reference: createReference(patient).reference,
+    display: getDisplayString(patient),
+  };
+
+  //Create a practitioner if if isn't already created
+  const practitioner = await medplum.createResourceIfNoneExist<Practitioner>(
+      {
+        resourceType: 'Practitioner',
+        id: 'practitioner',
+        name: [
+          {
+            given: ['Alice'],
+            family: 'Smith',
+          },
+        ],
+        photo: [
+          {
+            contentType: 'image/png',
+            url: 'https://docs.medplum.com/img/cdc-femaledoc.png',
+          },
+        ],
+      },
+      'Practitioner/practitioner'
+  );
+
+  patient.generalPractitioner = [createReference(practitioner)];
+  await medplum.updateResource(patient);
+
+  //Create a schedule for the practitioner
+  const schedule = await medplum.createResourceIfNoneExist<Schedule>(
+      {
+        resourceType: 'Schedule',
+        id: 'schedule',
+        actor: [
+          {
+            reference: createReference(practitioner).reference,
+            display: getDisplayString(practitioner),
+          },
+        ],
+      },
+      'Schedule/schedule'
```

Comment:
Same with this ID, something like Schedule/sched-12345

Reply:
The syntax here needs to be a search string, not by id.  You could search by identifier, such as `Schedule?identifier=xyz`